persistence/provider/sql/mysql/offsetstore.go

```diff
@@ -0,0 +1,81 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/dogmatiq/infix/eventstream"
+	"github.com/dogmatiq/infix/internal/x/sqlx"
+)
+
+// LoadOffset loads the last offset associated with the given source
+// application key sk. ak is the 'owner' application key.
+//
+// If there is no offset associated with the given source application key,
+// the offset is returned as zero and error as nil.
+func (driver) LoadOffset(
+	ctx context.Context,
+	db *sql.DB,
+	ak, sk string,
+) (eventstream.Offset, error) {
+	var o eventstream.Offset
+
+	row := db.QueryRowContext(
+		ctx,
+		`SELECT next_offset
+		FROM offset_store
+		WHERE app_key = ? AND source_app_key = ?`,
+		ak, sk,
+	)
+
+	if err := row.Scan(&o); err != nil {
+		if err == sql.ErrNoRows {
+			return 0, nil
+		}
+
+		return 0, err
+	}
+
+	return o, nil
```

Comment:
This is the other half of the suggestion at the top of this function that move the declaration of `o`. I'm also just trying to eliminate unnecessary branching, however minor. IMO this makes it clearer that in the context of this operation `ErrNoRows` is not an error.

```suggestion
	var o eventstream.Offset
	err := row.Scan(&o)
	if err == sql.ErrNoRows {
		err = nil
	}

	return o, err
```